### DISCUSSION TITLE
connecting xbox to VM problem

### DISCUSSION THREAD

**Discussion by kyoeun-space** [bitcraze/discussions#868] [Q&A]
![image](https://github.com/bitcraze/discussions/assets/122524815/7c0ec086-ba5f-408a-ba9b-fecb294c1ebe)

I'm trying to connect an xbox to VM. However, VM does not have an xbox connection. There was a very short connection, but I couldn't control the drone. I also installed an xbox driver. I don't know what the problem is. Is it just my laptop problem?

The picture is a capture of an uncontrollable xbox when it's connected for a very short time.

**Comment by krichardsson** [comment]
You say it worked for a short time? Do you know what you changed when it stopped working?
Which OS are you using?

> I also installed an xbox driver

Was this in the underlying OS on your machine?

**Reply by kyoeun-space** [reply]
The os I'm using is window and I'm running crazyflie as vm. I've repeated connecting and releasing the xbox usb cable countless times. Nevertheless, the connection was not made. I thought I might have to install the xbox driver on ubuntu, so I ran the same command as in the picture. (I don't think it helped) 

![image](https://github.com/bitcraze/discussions/assets/122524815/e67d5e7d-02fa-423f-8e02-f81b699b5e40)
![image](https://github.com/bitcraze/discussions/assets/122524815/2772ef1d-d51a-41db-9187-9f9168223821)


after installation, I repeated connecting and disconnecting xbox usb. at some point I dont even know, input device was detected, I tried to run "configure" by clicking "xbox-360" straight away without touching anything. But there was only a gray button like the picture. 

sadly, the drone's battery charging part fell off and I couldn't retry. I'm going to reboot my laptop as early as today and try again. If there is a problem again, I will come to report. Before that, I had a question.

Is it correct to choose xbox 360 for whatever xbox I use?
I was wondering if I had to click "joystick" or another.

thank you!

**Reply by krichardsson** [reply]
You need to let the VM know about the USB connection as well, gave you configured the USB port in the VM [as documented in the tutorial?](https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/#config-usb-vm)

You can find some more information in the [client documentation about controllers](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/inputdevices/), but I think you already did this

> Is it correct to choose xbox 360 for whatever xbox I use?

Yes, but it is really only a mapping of the buttons. You can change it any way you like as described in [the client documentation](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#creating-new-mappings)

**Reply by kyoeun-space** [reply]
yes, I gave configured the usb port in the vm.

The problem with the connection I've been finding for a few days seems to be related to the question I just posted. The m1 led blinks red several times. (https://github.com/orgs/bitcraze/discussions/877)

I look forward to a solution. Thank you for your help.

**Reply by krichardsson** [reply]
Yes, if there is a problem with the deck, the Crazyflie will not fly. 
If you want to verify that your game pad is working, you can remove the deck and fly the Crazyflie manually instead.
